read.go

```diff
@@ -50,6 +50,14 @@ func (r *reader) readTag() (*tag.Tag, error) {
 	if gerr == nil && eerr == nil {
 		return &tag.Tag{Group: group, Element: element}, nil
 	}
+	if gerr != nil || eerr != nil {
+		if gerr.Error() == "EOF" {
+			return nil, gerr
+		}
+		if eerr.Error() == "EOF" {
+			return nil, eerr
+		}
+	}
```

Comment:
Is the purpose of this to ensure readTag returns an io.EOF error instead of the compound error on L61?

One option once we target go1.20 is to use [errors.Join](https://pkg.go.dev/errors@master#Join) but in the meantime, perhaps the following would be more readable:

```suggestion
	// If the error is an io.EOF, return the error directly instead of the compound error later.
	// Once we target go1.20 we can use errors.Join: https://pkg.go.dev/errors#Join
	if errors.Is(gerr, io.EOF) {
		return nil, gerr
	}
	if errors.Is(eerr, io.EOF) {
		return nil, eerr
	}
```